gui/src/JobSelector.tsx

```diff
@@ -0,0 +1,711 @@
+                <Checkbox color="primary" checked={isItemSelected} onClick={(event) => doCheckboxClick(event, id)} />
+            </TableCell>
+            {displayJobId ? (
+                <TableCell component="th" id={id} scope="row" padding="none">
+                    <ListItemText primary={jobName} secondary={id} />
+                </TableCell>
+            ) : null}
+            {displaySubmitId ? <TableCell>{submitId}</TableCell> : null}
+            {displayEntityLabel || displayEntityId ? (
+                <TableCell>
+                    {!displayEntityLabel && displayEntityId ? (
+                        entityId
+                    ) : !displayEntityId && displayEntityLabel ? (
+                        entityName
+                    ) : (
+                        <ListItemText primary={entityName} secondary={entityId} />
+                    )}
+                </TableCell>
+            ) : null}
+            {displayDate ? <TableCell>{creationDate ? new Date(creationDate).toLocaleString() : ""}</TableCell> : null}
+            <TableCell>
+                <ChipStatus status={status} />
+            </TableCell>
+            <TableCell>
+                {status === JobStatus.RUNNING ? null : status === JobStatus.BLOCKED ||
+                  status === JobStatus.PENDING ||
+                  status === JobStatus.SUBMITTED ? (
+                    <IconButton onClick={() => doCancelJob([id])}>
+                        <StopCircleOutlined />
+                    </IconButton>
+                ) : (
+                    <IconButton onClick={() => doDeleteJob([id])}>
+                        <DeleteOutline color="primary" />
+                    </IconButton>
+                )}
+            </TableCell>
+        </TableRow>
+    );
+}
+
+const JobSelector = (props: JobSelectorProps) => {
+    const { id = "" } = props;
+    const [selected, setSelected] = useState<string[]>([]);
+    const [jobRows, setJobRows] = useState<Jobs>([]);
+    const [filters, setFilters] = useState<FilterData[]>();
+    const [anchorEl, setAnchorEl] = React.useState<HTMLButtonElement | null>(null);
+
+    const isSelected = (name: string) => selected.indexOf(name) !== -1;
+
+    const dispatch = useDispatch();
+    const module = useModule();
+
+    const className = useClassNames(props.className, props.dynamicClassName);
+
+    useDispatchRequestUpdateOnFirstRender(dispatch, id, module, props.updateVars);
+
+    const displayJobId = useDynamicProperty(props.displayJobId, props.defaultDisplayJobId, true);
+    const displayEntityLabel = useDynamicProperty(props.displayEntityLabel, props.defaultDisplayEntityLabel, true);
+    const displayEntityId = useDynamicProperty(props.displayEntityId, props.defaultDisplayEntityId, true);
+    const displaySubmitId = useDynamicProperty(props.displaySubmitId, props.defaultDisplaySubmitId, true);
+    const displayDate = useDynamicProperty(props.displayDate, props.defaultDisplayDate, true);
```

Comment:
These properties are not dynamic, they can simply be declared in the props spreading with their default value.
ie : Line 424 const { id = "", displayJobId = true, ...} = props;